gcs-action/src/gcs.ts

```diff
@@ -0,0 +1,202 @@
+import {
+  Storage,
+  UploadResponse,
+  DeleteFileResponse,
+  File
+} from '@google-cloud/storage'
+import * as fs from 'fs'
+import * as path from 'path'
+import * as core from '@actions/core'
+
+const storage = new Storage()
+
+/**
+ * Uploads a file to a bucket
+ * based on https://github.com/googleapis/nodejs-storage/blob/master/samples/uploadFile.js
+ * @param bucketName The name of the bucket
+ * @param filename The file path
+ * @param destination The destination prefix
+ */
+export async function uploadFile(
+  bucketName: string,
+  filename: string,
+  destination?: string
+): Promise<UploadResponse | Error> {
+  interface UploadOptions {
+    gzip: boolean
+    destination?: string
+  }
+  const options: UploadOptions = {gzip: true}
+  if (destination) {
+    options.destination = `${destination}${filename.slice(
+      filename.lastIndexOf('/') + 1
+    )}`
+  }
+  try {
+    const uploadedFile = await storage
+      .bucket(bucketName)
+      .upload(filename, options)
+    console.log(`${filename} uploaded to ${bucketName}.`)
+    return uploadedFile
+  } catch (err) {
+    return err
+  }
+}
+
+/**
+ * List all files in a bucket, filtered by prefix
+ * based on https://github.com/googleapis/nodejs-storage/blob/master/samples/listFilesByPrefix.js
+ * @param bucketName The name of the bucket
+ * @param prefix The prefix to filter by
+ * @param delimiter Optional delimiter
+ */
+export async function listFilesByPrefix(
+  bucketName: string,
+  prefix: string,
+  delimiter?: string
+): Promise<File[]> {
+  //interface for listing options
+  interface ListOptions {
+    prefix: string
+    delimiter?: string
+  }
+  const options: ListOptions = {prefix}
+
+  if (delimiter) {
+    options.delimiter = delimiter
+  }
+  // Lists files in the bucket, filtered by a prefix
+  const [files] = await storage.bucket(bucketName).getFiles(options)
+  return files
+}
+
+/**
+ * Deletes a file in a bucket
+ * based on https://github.com/googleapis/nodejs-storage/blob/master/samples/deleteFile.js
+ * @param bucketName The name of the bucket
+ * @param filename The name of the name of the file to be deleted
+ */
+export async function deleteFile(
+  bucketName: string,
+  filename: string
+): Promise<{status: DeleteFileResponse}> {
+  try {
+    const deletedFile = await storage
+      .bucket(bucketName)
+      .file(filename)
+      .delete()
+    console.log(`gs://${bucketName}/${filename} deleted.`)
```

Comment:
If we're trying to send messages back to the user, we need to use `actions/core`.

Reply:
Oops, creeped in from when I was testing this.